src/OpenSpaceToolkit/Astrodynamics/Flight/System/PropulsionSystem.cpp

```diff
@@ -0,0 +1,116 @@
+/// Apache License 2.0
+
+#include <OpenSpaceToolkit/Core/Error.hpp>
+#include <OpenSpaceToolkit/Core/Utilities.hpp>
+
+#include <OpenSpaceToolkit/Physics/Environment/Gravitational/Earth.hpp>
+
+#include <OpenSpaceToolkit/Astrodynamics/Flight/System/PropulsionSystem.hpp>
+
+namespace ostk
+{
+namespace astro
+{
+namespace flight
+{
+namespace system
+{
+
+using ostk::physics::environment::gravitational::Earth;
+
+PropulsionSystem::PropulsionSystem(const Scalar& aThrust, const Scalar& aSpecificImpulse)
+    : thrust_(aThrust),
+      specificImpulse_(aSpecificImpulse),
+      massFlowRate_({aThrust.getValue() / (aSpecificImpulse.getValue() * Earth::gravityConstant), massFlowRateSIUnit_})
```

Comment:
Can we do this blindly? Should we ensure that `aThrust` and `aSpecificImpulse` values are transformed to the right units beforehand?

Reply:
Good point, I am adding check to throw errors on this for now